blockchain/scriptval.go

```diff
@@ -142,8 +142,10 @@ func (v *txValidator) Validate(items []*txValidateItem) error {
 
 	// Start up validation handlers that are used to asynchronously
 	// validate each transaction input.
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
```

Comment:
This code is in a super hot path.  There are also no early exists.  Please move this down where the `close(v.quitChan)` is currently just before the `return nil` to avoid the defer.